crates/bevy_ecs/src/system/system_param.rs

```diff
@@ -356,6 +365,140 @@ fn assert_component_access_compatibility(
     panic!("error[B0001]: Query<{query_type}, {filter_type}> in system {system_name} accesses component(s){accesses} in a way that conflicts with a previous system parameter. Consider using `Without<T>` to create disjoint Queries or merging conflicting Queries into a `ParamSet`. See: https://bevyengine.org/learn/errors/b0001");
 }
 
+// SAFETY: Relevant query ComponentId and ArchetypeComponentId access is applied to SystemMeta. If
+// this Query conflicts with any prior access, a panic will occur.
+unsafe impl<'a, D: QueryData + 'static, F: QueryFilter + 'static> SystemParam
+    for QuerySingle<'a, D, F>
+{
+    type State = QueryState<D, F>;
+    type Item<'w, 's> = QuerySingle<'w, D, F>;
```

Comment:
Why isn't this just `<D as QueryData>::Item`? I think that would improve the ergonomics and complexity quite a bit?

Reply:
The ergonomics comment got removed, CI told me to remove unnecessary syntax which made it way better.

`<D as WorldQuery>::Item` is not a `SystemParam` so that's one issue with this.
Another would be, if `SystemParam` returns anything else than itself with changed lifetimes, it won't fit as the system argument.
```rs
fn my_system(q: QuerySingle<Entity>) { ... }

// imaginary executor code
let q: <Entity as WorldQuery>::Item = QuerySingle<Entity>::get_param(world);
system(q); // Expected `QuerySingle` got `<Entity as WorldQuery>::Item`
```